src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,125 @@
+import {
+  memo, useCallback, useEffect, useState,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { SexCategory } from '../types/SexCategory';
+
+const findParent = (people: Person[], parentName: string | null) => {
+  return people.find(parent => parent.name === parentName);
+};
+
+export const PeoplePage = memo(() => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loadingError, setLoadingError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [searchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex') || SexCategory.NONE;
+  const centuries = searchParams.getAll('centuries') || [];
+  const query = searchParams.get('query') || '';
+
+  const loadPeople = useCallback(async () => {
```

Comment:
As you use this function only inside the `useEffect` you may move it there. Don't just use `useCallback` without a reason.

<img width="910" alt="image" src="https://github.com/mate-academy/react_people-table-advanced/assets/102293862/eb7f1593-2bbc-4fe6-a8e5-8ff9cd71d6bc">


https://react.dev/reference/react/useCallback